kb/2022-03-10/multiple_nics_vm_connectivity.md

```diff
@@ -0,0 +1,148 @@
+
+## How-to identify connectivity issues
+
+In a Linux VM, you can use commands from the `iproute2` package to identify the default route.
+
+In your VM, execute the following command:
+```bash
+ip route show default
+```
+!!! note
+    If you get the `access denied` error, please run the command using `sudo`
+    
+The command's output will display only the `default` route with the gateway and VM IP of the main network interface (`eth0` in the example below).
+```
+default via <Gateway IP> dev eth0 proto dhcp src <VM IP> metric 100
+```
+
+Here is the full example:
+```
+$ ip route show default
+default via 192.168.0.254 dev eth0 proto dhcp src 192.168.0.100 metric 100
+$
+```
+
+However, if the issue covered in this KB occurs, you'll only be able to connect to the VM via the Harvester Web Console.
+
+Once connected, you can run again the same command as before:
+```bash
+ip route show default
+```
+
+However, this time you'll get a default route with an incorrect gateway IP.
+For example:
+```
+default via <Incorrect Gateway IP> dev eth0 proto dhcp src <VM's IP> metric 100
+```
+
+## Why does connectivity issues occurs randomly
+
+In a standard setup, cloud based VMs are very often configured with DHCP for all NICs by default. The configuration sets an IP and a gateway for each NIC. Lastly, a default route to the gateway IP will also be added, so you can connect to the VM using its IP.
+
+However, Linux distributions start multiple DHCP clients simultaneously and don't have a "prioritization" system. This means that in the case you have two or more NICs configured with DHCP, the clients will enter a "race condition" to configure the default route. And depending on the Linux distribution DHCP script currently running, there's no guarantee on which default route will be configured.
+
+As the default route might change in every DHCP renewing process or after every OS reboot, this will create the network connectivity issues.
+
+## How to avoid the random connectivity issues
+
+You could avoid these connectivity issues easily by having only one NIC attached to the VM and have only one IP and one gateway configured.
+
+However, the use of only one NIC is often not possible for VMs in a more complex infrastructure. For example, if your infrastructure has a storage network and a service network. For security reasons, the storage network will be isolated from the service network, and have an independent subnet. In this situation, you must have two NICs to connect to both the service and storage networks.
+
+As a solution, you can choose one solution from below to fit your requirement and security policy.
+
+### Disable DHCP on secondary NIC
+
+As explained above, the issue is caused by a "race condition" of two DHCP clients. One solution to avoid this issue, is to disable DHCP for the all NICs and configure them only with a static IP. In the same way, you could configure the secondary NIC with a static IP and keep the primary NIC with DHCP enabled.
```

Comment:
```suggestion
As mentioned above, the problem is caused by a `race condition` between two DHCP clients. One solution to avoid this problem is to disable DHCP for all NICs and configure them with static IPs only. Likewise, you can configure the secondary NIC with a static IP and keep the primary NIC enabled with DHCP.
```